### PR TITLE
[v1.16] CODEOWNERS: switch cilium/tophat to cilium/committers

### PR DESCRIPTION
**Pull request by julianwiedmann**
Membership in `tophat` typically changes on a weekly basis, resulting in stale review approvals after a change-over. It's also a single source of failure in our review & merge machinery.

As tophat for stable branches is primarily meant to review for process conformity (backport criteria, correct formatting of the PR etc), pull in all of the committers instead to perform this important responsibility.

Note: if you can't find a single committer to approve your stable PR, it most likely shouldn't be merged.

### CODE REVIEW

**Comment by michi-covalent** [comment]
i guess we need to somehow ensure all the committers apply the backporting criteria consistently https://docs.cilium.io/en/stable/contributing/release/backports/#id1

at some point i also wondered if the default code owners for stable branches should be @cilium/release-managers 💭 let's see how they feel about that.

another option might be to not remove people from the tophat team. we just need to configure review assignments right?

**Review by joestringer — APPROVED** [review]
I don't love assigning ownership to committers mainly due to the bystander effect - a large group is responsible, so no-one is responsible. I agree in principle that you should be able to find at least one committer to "bless" your PR, but on the other hand one of the key properties of the rest of the codeowners system is that exactly one person gets assigned, so they are responsible to either respond or reach out for help from the rest of the codeowner group to respond. Even with tophat at least there's typically max 2 people it could be, so it's pretty quick to find out exactly who should take a look. All that said, I agree with you that the constantly rotating tophat doesn't really work today.

In terms of maintaining stability of older branches, ultimately I think it is up to the @cilium/committers, so unless we opt to delegate that in some way, I think this is a reasonable approach to ensure that anyone who is looking to make changes to stable branches only needs sign off from any someone in the core team.

We may want to think about whether we form something a bit more formal for stable maintenance, whether that's:
- A subset of the committers for all stable releases (could even be @cilium/release-managers ), or
- A group based on contributors actively involved in backporting, or
- New stable teams for each Cilium release (v1.16, v1.17, ...) - At the start of the maintenance cycle we assemble a new team of volunteers who will keep an eye on that stable branch for the following 6 months of active maintenance + 12 following months of security fix maintenance.
- Some other variation on this theme

Either way in the immediate term I think this is an improvement to the process, and is worth trying out. If we find that it's painful as well, then that gives us good motivation to formalize this a bit more.

Probably worth making this change for 1.15, 1.14 if we're in agreement here.

**Review by youngnick — APPROVED** [review]
Seems fine from under the tophat to me.

**Comment by YutaroHayakawa** [comment]
> I don't love assigning ownership to committers mainly due to the bystander effect - a large group is responsible, so no-one is responsible.

I agree with the Joe's opinion here, but I guess it's a matter of configuring the review assignment? I'm fine with taking care of the review explicitly assigned to me, but otherwise, I guess I wouldn't take the responsibility of reviewing the PR out of my expertise out of 30 people. I would think there's sufficient people than me.

**Comment by ti-mo** [comment]
As a member of the tophat rotation, I'm really not looking forward to getting pinged for every single backport PR that gets created (as opposed to only getting pinged on my shift). If Yutaro's response is any indication, bystander effect will be huge, quite expected when blasting these out to 30 people.

IMO Joe's proposal is better, letting backport merges be coordinated by other backporters and/or release managers would be more effective. Release managers currently need to call for merges to be paused during branch preparation, and this likely causes some stress on their part if they feel like they're holding people up. Putting them in control of the whole process would cause less friction, if the workload is manageable, of course.

**Comment by joestringer** [comment]
Something else that's probably worth noting is that the common case is there's already at least one committer reviewing backport PRs. If committers are the codeowner then this will naturally resolve the requirement for review. We don't necessarily have to directly assign a particular committer. In some ways this proposal as-is just makes the GitHub review system for stable branches more in line with how we actually review stable PRs - best effort review and sanity check from authors, then in the end committers (including backporters/release-managers) have rights to merge the PR.

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS:228]: I realized that the other common pattern for PRs on stable branches is when preparing a release. We may want to consider adding them in as well as optional approvers:

```suggestion
* @cilium/committers @cilium/release-managers
```

Comment (aanm) [reply CODEOWNERS:228]: Let's see how things work out without `@cilium/release-managers`

**Review by aanm — APPROVED** [review]

**Pull request merged**
